src/classes/RD2_RecurringDonation.cls

```diff
@@ -204,21 +209,70 @@ public inherited sharing class RD2_RecurringDonation {
 
     /**
     *@description Revises and modifies the Recurring Donation if it is Open Ended and it past the RD End Date
+    * @param currentDate Used to determined current date
     * @return RD2_RecurringDonation This record instance
     */
-    public RD2_RecurringDonation reviseStatusForOpenEnded(Date currentDate) {
-        if (isActive()
-            && isOpenEnded()
+    private RD2_RecurringDonation reviseStatusForOpenEnded(Date currentDate) {
+        if (isOpenEnded()
             && rd.EndDate__c != null
             && rd.EndDate__c < currentDate
         ) {
             rd.Status__c = RD2_Constants.STATUS_CLOSED;
-
             changedFields.add(npe03__Recurring_Donation__c.Status__c);
+
         }
+
         return this;
     }
 
+    /**
+    * @description Revises and modifies for paused and unpaused Recurring Donation
+    * @param currentDate Used to determined current date
+    * @param scheduleService RD Schedule Service
+    */
+    private void reviseStatusForPaused(Date currentDate, RD2_ScheduleService scheduleService) {
+        if (getState() == RD2_Constants.STATUS_CLOSED
+            || rd.RecurringDonationSchedules__r == null
+        ) {
+            return;
+        }
+   
+        if (!scheduleService.pauseHandler.hasActivePause(rd.RecurringDonationSchedules__r, currentDate)) {
+            if (rd.Status__c == RD2_Constants.STATUS_PAUSED) {
+                rd.Status__c = RD2_Constants.STATUS_ACTIVE;
+                rd.ClosedReason__c = '';
+                changedFields.add(npe03__Recurring_Donation__c.Status__c);
+                changedFields.add(npe03__Recurring_Donation__c.ClosedReason__c);
+            }
+            return;
+        }
+
+        Boolean isPauseStatusValid = UTIL_Describe.isValidPicklistOption(
+            'npe03__Recurring_Donation__c',
+            String.valueOf(npe03__Recurring_Donation__c.Status__c),
+            RD2_Constants.STATUS_PAUSED);
+        
+        if (!isPauseStatusValid) {
+            return;
+        }
+
+        RecurringDonationSchedule__c nextPauseSchedule = scheduleService.getNextActivePauseSchedule(rd.RecurringDonationSchedules__r);
```

Comment:
This call and the installments are using the `scheduleService`, meaning the code can be extracted into the `scheduleService` and answer question `scheduleService.isCurrentPause(rd.RecurringDonationSchedules__r, currentDate)`, the method should probably be defined in the `pauseHandler` after `hasActivePause()`. Probably do not need to call the `scheduleService.pauseHandler.hasActivePause()` on line 240 at all, just check `isCurrentPause()`. If not and the Status is Pause, then revert to Active. If the RD is currently paused, and the Status picklist value is valid, then set the RD to paused if not already set.